### ISSUE TITLE
link rel=preconnect does not work in Firefox

### ISSUE THREAD

**Issue by jakub-g** [Fyrd/caniuse#5281]
https://caniuse.com/#feat=link-rel-preconnect

I opened a bug > 1 year ago and it's still not fixed:
https://bugzilla.mozilla.org/show_bug.cgi?id=1543990

It regressed at some point in the past but not clear when exactly

**Comment by aesyondu** [comment]
I'm not sure how to read this waterfall thing, but based on your link, if it's not doing DNS resolution, etc., then it seems to be fixed

<details>
<summary>
Firefox Dev 74.0b2:
</summary>
<img width="670" alt="image" src="https://user-images.githubusercontent.com/57334943/74489228-9d3eff80-4eff-11ea-9267-67698bb99ad7.png">
</details>

<details>
<summary>
Firefox 72.0.2:
</summary>
<img width="497" alt="image" src="https://user-images.githubusercontent.com/57334943/74489401-25bda000-4f00-11ea-96dc-e94e7a8d7d03.png">
</details>

My issue is why is there is duplicate preconnect entry, which one is correct?

<img width="722" alt="image" src="https://user-images.githubusercontent.com/57334943/74489630-c744f180-4f00-11ea-8788-7a85e2258b22.png">

**Comment by jakub-g** [comment]
Hi @aesyondu thx for looking into this.

Regarding your test, I redid it in stable72 and nightly75 on Windows and still see the bug (that is, the <img> request has DNS,TCP,TLS phases in Timings section.

The test methodology is important: if you repeat the test (refresh the page), then those steps won't happen anymore, because the browser maintains the connection for some number of seconds.

The ideal test methodology is:

1. Close the test URL tab (to make sure it doesn't reload automatically on browser restart)
2. Close the browser (to clean the connection pool)
3. Optionally, clear DNS cache of the operating system
4. Reopen the browser
5. Open a new tab
6. Open devtools (to make sure you capture all the network requests)
7. Load test page

Good point about having duplicated entries, IMO the second one could be nuked probably? (and notes about crossorigin can be transferred to the first one)

**Comment by aesyondu** [comment]
Thanks @jakub-g! I redid it as you suggested and replicated the bug:

<img width="684" alt="image" src="https://user-images.githubusercontent.com/57334943/74525130-61398800-4f5b-11ea-8c25-79f8618cedbc.png">

On second reload, it does store the connection somehow:

<img width="675" alt="image" src="https://user-images.githubusercontent.com/57334943/74525247-9b0a8e80-4f5b-11ea-8a1b-9df3df9c1e07.png">

I'm on Max OS X 10.14.6

**Comment by jakub-g** [comment]
Yep it seems that Firefox keeps the connection open for 115 seconds
http://kb.mozillazine.org/Network.http.keep-alive.timeout
(verified in `about:config` this is still the value used in v72)
